numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -34,6 +34,16 @@
 
 #include "binop_override.h"
 
+/*
+ * used for allocating a single scalar, so use the default numpy
+ * memory allocators instead of the (maybe) user overrides
+ */
+NPY_NO_EXPORT void *
+npy_alloc_cache_zero(size_t nmemb, size_t size);
+
+NPY_NO_EXPORT void
+npy_free_cache(void * p, npy_uintp sz);
```

Comment:
I honestly wonder if we should not just use the Python allocator for these scalar-sized buffers.  But just nitpick/suggestion I am happy to keep it as is for now.